feat: add floating docs navigation shortcuts

Yash-kurosaki opened the pull request:
This PR implements the floating docs navigation shortcuts as requested in issue #4629.

### Summary
- Added `DocsNavigationShortcuts` component
- Integrated shortcuts into `DocsLayout.tsx`
- Ensures keyboard navigation is accessible and visible
- Tested locally using `npm run dev`

here is the screenshot :
<img width="1280" height="691" alt="image" src="https://github.com/user-attachments/assets/a3ba43f9-1976-491e-8564-4d3decb0842e" />

### Issue Reference
Fixes #4629

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added keyboard shortcuts (left/right arrow) to navigate between documentation pages for faster browsing.
* **Style**
  * Updated documentation navigation button container with a subtle sticky, blurred background and rounded card styling for improved visibility and UX.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Yash-kurosaki committed:
feat: add floating docs navigation shortcuts

bandantonio commented:
I would vote **against** having floating buttons that overlap with the main content. This is kinda awful UX.
Furthermore, the original issuer didn't mention that the buttons must be floating.

Yash-kurosaki committed:
fix: remove floating buttons, keep keyboard shortcuts only

- Removed floating navigation buttons that overlapped content (bad UX)
- Kept keyboard shortcuts (← → arrow keys) for quick navigation
- Made existing DocsButton sticky for better accessibility
- Fixed linter warnings (consistent-return, cleanup formatting)

Addresses maintainer feedback: floating buttons removed, keyboard navigation preserved.

Yash-kurosaki commented:
> I would vote **against** having floating buttons that overlap with the main content. This is kinda awful UX. Furthermore, the original issuer didn't mention that the buttons must be floating.

Thanks @bandantonio for the feedback. I've updated the PR to address your concerns

Yash-kurosaki commented:
Hlo maintainers,
Just a gentle ping on this PR.
I’ve addressed the UX feedback by removing floating buttons and keeping keyboard navigation.
Let me know if any further changes are needed. Thanks!

princerajpoot20 commented:
i too agree we don't need this.
FYI: https://github.com/asyncapi/website/issues/4629#issuecomment-4184177784

thanks for your effort @Yash-kurosaki
